### ISSUE TITLE
Allow for one to many Command translation 💭 

### ISSUE THREAD

**Issue by Nolski** [OpenMined/threepio#52]
# Summary

Some commands in one framework don't have direct maps to a single command in another framework. We need to support chaining multiple commands together to allow for custom implementations between frameworks.

# Background

**Is the improvement related to a problem? If yes, please describe**:

For example `torch.nn.functional.linear` doesn't have a directly mapped command in tensorflow.js. The source code shows `linear` to essentially be the below:
```python
def linear(input, weight, bias = None):
    output = input.matmul(weight.t())
    if bias is not None:
        output += bias
    return output
```

**Describe the ideal solution**:

Threepio will translate one command into many. This may change the output away from something like `Command` and into something resembling a PySyft `Plan` object. `torch.nn.functional.linear` would transform into a series of commands similar to below:

```
[
  {name: 't', input: 'weight', output: 'weight_o', ...},
  {name: 'matmul', input: ['input', 'weight_o'], 'output': 'output' ...},
  {name: 'add', input: [output, bias], output: output_o ...},
]
```

**Describe alternatives considered**:

We could modify syft.js to look up the keys of commands prior to executing them and executing corresponding blocks of custom translated code. This is a bit more difficult to maintain but will be less complex than extending the functionality of `Plan` to contain a `Plan` as an item within `Plan.actions`.

We could also work to simply patch frameworks with API endpoints that can directly translate instead of performing the translation ourselves.

# Outcome

Threepio supporting translations for commands that don't have a direct translation.

**Comment by tudorcebere** [comment]
This looks awesome! If you have any ongoing PR, please link it, if not, I will add the available tag on this. This looks like a cool solution to this problem. @Jasopaum, @vvmnnnkv any thoughts about this?

**Comment by Jasopaum** [comment]
Yep, this looks nice and I'd be very interested in seing the implementation of that too 🙂 
My only concern is that it might be a lot of work to specify the translations for all the commands between all the different frameworks, right? Or maybe I just don't see how you would automatize it

**Comment by Nolski** [comment]
@tudorcebere Thanks! So the last bits of general scaffolding needed to support multi-command translation are in review on #88. `Threepio.translate()` now returns a list of `Command`s. You can add custom translations in our [translations.py file](https://github.com/OpenMined/threepio/blob/67d0a51bff01abb59c74428fec30724a52077891/docs-crawler/docs/translations.py#L14). I have yet to implement the functionality which allows Commands to mark output variables and have them map as input to following commands. That will hopefully be coming this week :)

@JasoPaum It's definitely been a lot of work :joy:  but it's almost there! Many commands are automatically mapped based on their [function name](https://raw.githubusercontent.com/OpenMined/threepio/67d0a51bff01abb59c74428fec30724a52077891/pythreepio/static/mapped_commands_full.json) and threepio also allows for [word-level translation](https://github.com/OpenMined/threepio/blob/master/docs-crawler/docs/translations.py#L8) (for example, `gt` in pytorch will always map to `greater` in tensorflow.js). Lastly for commands which don't exist in certain frameworks, we can do a custom translation using our[ json schema](https://github.com/OpenMined/threepio/blob/master/docs-crawler/docs/translations.py#L8) this is how I plan to do one-to-many command translations.

**Comment by Nolski** [comment]
Resolved by #89
